plugins/grpc/instrument.go

```diff
@@ -67,6 +67,21 @@ func (i *Instrument) Points() []*instrument.Point {
 				instrument.WithResultType(0, "ClientStream")),
 			Interceptor: "ClientStreamingInterceptor",
 		},
+		{
+			PackagePath: "",
+			At: instrument.NewMethodEnhance("*ClientConnInterface", "Invoke",
```

Comment:
The interface cannot be intercepted, and I think intercepting `ClientConnInterface` is not effective.

`ClientConn` implements this interface, and when user make RPC calls, the Invoke and `NewStream` methods implemented by `ClientConn` are executed.

Reply:
Let me do more tests. I didn't get the tracing data of gRPC before. Below is my usage.

https://github.com/LinuxSuRen/api-testing/blob/58905fdd8f42b9f988f6312c7fe72034e0e12fc7/pkg/runner/grpc.go#L329

Reply:
Sure. I need more time to make sure my use case is correct. I will add the corresponding tests after that.

Reply:
I feel very confused. The e2e test seem correct, but it cannot work in my case. So, I tend to close this PR until I can figure it out.

By the way, I didn't find a way to control if sending the data to SK server or not. For some use cases (at least in my cas), SkyWalking is not required for the installation. So, I'm wonderfing if there is a way to control it.

I found a possible solution is that. Read an environment variable `SW_AGENT_ENABLED`, return `false` if its value is `false`.

https://github.com/apache/skywalking-go/blob/b3e0cf3c66e3a4d9010d050b6a2015c4014b8a33/tools/go-agent/instrument/reporter/instrument.go#L47

Reply:
> By the way, I didn't find a way to control if sending the data to SK server or not

No, you can't. That is kernel responsibility, including connection management, TLS, traffic limitation, etc.

> For some use cases (at least in my cas), SkyWalking is not required for the installation

What do you mean some cases? Do you mean some URL or protocol? Or something else? You want to compile the agent into your binary, but not using it?

Reply:
> What do you mean some cases? Do you mean some URL or protocol? Or something else? You want to compile the agent into your binary, but not using it?

Yes. Users might only want to install the tool instead of other optional component. For example my test tool, the following could start a server:

`atest server`

It will output the following errors if no SK server given. I'm thinking about if it's possible to avoid printing the errors.

`skywalking-go 2023/09/25 14:58:14 open stream error rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 0.0.0.0:11800: connect: connection refused"`

Reply:
In the previous discussion, we were not considering a general use tool would adopt this agent. So, the old proposal is going to package two binaries, one with the agent, the other is not. This should be good enough for a business service.

@mrproliu Do we have the mechanism for the runtime configuration for now? If so, maybe we could consider add one to disable the agent.

Reply:
Thanks for your explanation. Consider my project is very small and simple, having mutiple distribution is not a good choice. As you mentioned, it's not problem for those business vendors.

Reply:
Let's focus on the feature at here.